modules/engines/engine-xsjs/src/main/java/com/sap/xsk/synchronizer/XSJSLibSynchronizerUnpublisher.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2022 SAP SE or an SAP affiliate company and XSK contributors
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-FileCopyrightText: 2022 SAP SE or an SAP affiliate company and XSK contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.sap.xsk.synchronizer;
+
+import com.sap.xsk.synchronizer.XSJSLibSynchronizerPathTypeResolver.ResolvedPathType;
+import com.sap.xsk.synchronizer.cleaners.XSJSLibSynchronizerCleaner;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class XSJSLibSynchronizerUnpublisher {
+  private static final Logger logger = LoggerFactory.getLogger(XSJSLibSynchronizerUnpublisher.class);
+
+  private final XSJSLibSynchronizerPathTypeResolver resolver;
+
+  private final XSJSLibSynchronizerCleaner fileCleaner;
+
+  private final XSJSLibSynchronizerCleaner dbCleaner;
+
+  XSJSLibSynchronizerUnpublisher(
+      XSJSLibSynchronizerPathTypeResolver resolver,
+      XSJSLibSynchronizerCleaner fileCleaner,
+      XSJSLibSynchronizerCleaner dbCleaner
+  ) {
+    this.resolver = resolver;
+    this.fileCleaner = fileCleaner;
+    this.dbCleaner = dbCleaner;
+  }
+
+  public void unpublish(String targetRegistryPath) {
+      ResolvedPathType type = resolver.resolveWithCollectionFirst(targetRegistryPath);
+
+      switch (type) {
+        case EXISTENT_XSJSLIB_FILE: {
+          dbCleaner.cleanup(targetRegistryPath);
+          fileCleaner.cleanup(targetRegistryPath);
+        }
+        break;
+
+        case EXISTENT_FOLDER:
+          dbCleaner.cleanup(targetRegistryPath);
```

Comment:
Shouldn't you clean folder files too?

Reply:
When you unpublish a signle resource for example abc.xsjslib - the registry resource abc.xsjslib is deleted by the publisher processor or whatever handles unpublish, but we need to also delete the reserved name `abc.xsjslib.generated_exports` file. That is why we delete it on single resource unpublish.

As for your question it is related to the unpublishing of directories:
- No, the PR relies on the fact that when you unpublish a directory it deletes the entire directory from the registry. Otherwise we would have to recur inside that directory and find all the `*.xsjslib.generated_exports` files and delete them and it is useless as the unpublisher already does that by deleting the entire registry collection that they are contained in.